GitCommands/UserRepositoryHistory/RepositoryXmlSerialiser.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Xml.Serialization;
+using JetBrains.Annotations;
+
+namespace GitCommands.UserRepositoryHistory
+{
+    /// <summary>
+    /// Provides the ability to serialise and deserialise collections of user's git repositories.
+    /// </summary>
+    public interface IRepositorySerialiser
```

Comment:
:bulb: Separate file for top-level types

Reply:
I typically place an interface above a class that implements it. In rare cases where an interface has multiple implementations, the interface is placed in its own class.
The main reason for this organisation is that interfaces are closely related to the implementations (or vice versa), and are primarily created for an abstraction (i.e. decoupling) and testability reasons (i.e. SOLID).
From my experience, having interfaces in separate files creates more (visual) clutter and introduces a maintenance overhead.